File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import java.util.List;

import com.pinyougou.pojogroup.Cart;

public interface CartService {
	
	/**
	 * 	添加商品到购物车
	 * @param cartList
	 * @param itemId
	 * @param num
	 * @return
	 */
	public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
	
	/**
	 *	从Redis中提取购物车列表 
	 * @param username
	 * @return
	 */
	public List<Cart> findCartList(String username);
	/**
	 * 	将购物车存入Redis
	 * @param username
	 * @param cartList
	 */
	public void saveCartListRedis(String username,List<Cart> cartList);
	
	/**
	 * 	合并购物车
	 * @param cartList1
	 * @param cartList2
	 * @return
	 */
	public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function($http){
	//查询购物车列表
	this.findCartList=function(){
		return $http.get('cart/findCartList.do');
	}
	//添加商品到购物车
	this.addGoodsToCartList=function(itemId,num){
		return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
	}
	
	//求购物车商品合计
	this.sum=function(cartList){
		var totalValue={totalNum:0,totalMoney:0}
		
		for(var i=0;i<cartList.length;i++){
			var cart=cartList[i];//购物车对象
			for(var j=0;j<cart.orderItemList.length;j++){
				var orderItem=cart.orderItemList[j];
				totalValue.totalNum+=orderItem.num;
				totalValue.totalMoney+=orderItem.totalFee;
			}
		}
		
		return totalValue;
	}
	
	//获取地址列表
	this.findAddressList=function(){
		return $http.get('address/findListByLoginUser.do');	
	}
	
	//提交订单
	this.submit=function(order){
		return $http.post('order/add.do',order)
	}
	
})